src/types/markers.js

```diff
@@ -7,6 +7,49 @@ import type { Milliseconds, Microseconds, Seconds, Bytes } from './units';
 import type { GeckoMarkerStack } from './gecko-profile';
 import type { IndexIntoStackTable, IndexIntoStringTable } from './profile';
 
+// Provide different formatting options for strings.
+export type MarkerFormatType =
+  | 'string' // "Label: Some String"
+  | 'bytes' // "Label: 5mb"
+  | 'seconds' // "Label: 5s"
```

Comment:
Would only having `seconds` and just automatically formatting `.001` as `1ms` for example be reasonable?

Reply:
I agree. Furthermore, it would be good to have a distinction between time durations and points of time. For example, when picking a video frame during compositing, there are three timestamps involved: The time at which the function that does the picking is called, and the timestamp of the video frame that is picked, and the estimated output timestamp of the composition for which the video frame is picked. It would be nice to supply the latter two timestamps as marker fields.

Reply:
> only having `seconds`

I would mildly disagree with that one: I can imagine that some markers know that they're dealing with durations in certain expected ranges, so they should be able to always show the same unit for consistency.

> automatically formatting `.001` as `1ms` for example

> distinction between time durations and points of time

But these are both good ideas! So I think we should have two other format types: `duration` and `timestamp`, which would be displayed with the most appropriate unit; and `timestamp`s could have an `@` prefix to show they're points in time?
`...seconds` would still be there as durations with a set unit.

Reply:
I think it's useful to leave the UI decide most of the time, so we could have eg "autoseconds". I also agree that sometimes we could force a specific unit for consistency, I know padenot told me something about that too.

Now that I'm sold about integer values, I believe the values themselves should be in nanoseconds always.

Reply:
Thanks for all the feedback, I will attempt to incorporate it into future work. I'll focus on being permissive on added schema formatting options, as long as they aren't too complicated.